responder/routes.py

```diff
@@ -85,6 +86,10 @@ def matches(self, scope):
         if scope["type"] != "http":
             return False, {}
 
+        # if we specified allowed methods and this is not one of them, we don't match
+        if self.methods is not None and scope["method"] not in self.methods:
+            return False, {}
+
```

Comment:
Instead of doing the check here, do it when handling the request (__call__) and return a [method not allowed 405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)